### PR TITLE
0.3.0 socket

### PR DESCRIPTION
**Pull request by nshaaban-cPacket**
Moved contents of transport into socket. While there are more changes to be made in socket, it needs to be moved first before diffs are readable.

### CODE REVIEW

File: src/engineio/socket.rs [hunk 1]
```diff
@@ -1,22 +1,86 @@
...
+};
+use websocket::{
+    client::sync::Client as WsClient,
+    header::Headers,
+    sync::stream::{TcpStream, TlsStream},
+    ClientBuilder as WsClientBuilder,
+};
+use websocket::{
+    dataframe::Opcode, receiver::Reader, sync::Writer, ws::dataframe::DataFrame, Message,
+};
```
Comment (1c3t3a) [inline src/engineio/socket.rs:29]: Can't we union those imports? Or those the linter then starts to complain?

**Review by 1c3t3a — APPROVED** [review]
Looks good

File: src/engineio/socket.rs (hunk 1 above)
Comment (nshaaban-cPacket) [reply src/engineio/socket.rs]: done.

**Comment by nshaaban-cPacket** [comment]
> Looks good! I think this one should be merged before the others, am I right?

Yup! Awaiting your merge, I don't have permission.

**Pull request merged**